packages/babel-generator/src/generators/types.ts

```diff
@@ -3,9 +3,27 @@ import { isAssignmentPattern, isIdentifier } from "@babel/types";
 import type * as t from "@babel/types";
 import jsesc from "jsesc";
 
+let lastRawIdentNode: t.Identifier | null = null;
+let lastRawIdentResult: string = "";
+export function _getRawIdentifier(this: Printer, node: t.Identifier) {
+  if (node === lastRawIdentNode) return lastRawIdentResult;
+  lastRawIdentNode = node;
+
+  const { name } = node;
+  const token = this._findToken(tok => tok.value === name);
```

Comment:
Maybe we should also compare positions? If there are multiple tokens with the same value.